pkg/cdi/container-edits.go

```diff
@@ -220,3 +286,72 @@ func ensureOCIHooks(spec *oci.Spec) {
 		spec.Hooks = &oci.Hooks{}
 	}
 }
+
+// fillMissingInfo fills in missing mandatory attributes from the host device.
+func fillMissingInfo(dev *oci.LinuxDevice) error {
+	if dev.Type != "" && (dev.Major != 0 || dev.Type == "p") {
```

Comment:
So if the device type and major are set we return. Should we also check the minor number?

Reply:
No. IMO, it does not make sense to omit the minor number. So you either omit both or none. Even if we tried, with the current struct we can't know if a 0 minor number is explicitly set to 0 (which is valid) or omitted, because the structure has the member as `Minor int`, not `Minor *int`.